### PR TITLE
[WM-2255] Fix small JSON parser bug for workflow inputs

### PR DESCRIPTION
**Pull request by rickymagner**
## Ticket
https://broadworkbench.atlassian.net/browse/WM-2255

## Summary of changes:
When saving/re-uploading configurations for workflows, there's a small bug that causes the inputs to be incorrectly parsed when a variable is an array containing newlines. This short PR resolves this issue by correcting the regex used to parse JSON files here.

### What
When saving a multi-line array inside your workflow inputs like below
image
and then re-uploading, it gets incorrectly parsed like so:
image

### Why
This was previously occurring due to a small quirk in JS regex, where `.` matches any character *except* newlines. This has been replaced with the more generic, newline-matching-inclusive regex `[\s\S]` so that multi-line arrays like the above get matched with the built-in regex matcher and correctly parsed, as opposed to being dumped into a generic "not sure what this is" string. It seems there are other expressions for "match all characters including newline" for JS, but after reading around a bit, it seemed like this was one of the most universally recognized across browsers.

### Testing strategy
I tried it out with my local `yarn start` instance.

### CODE REVIEW

**Review by cjllanwarne — APPROVED** [review]
Looks correct to me. Would love to see this become a testable function with some unit tests (eg in `WorkflowView.test.js`) to prove that everything is being matched and replaced correctly

File: src/pages/workspaces/workspace/workflows/WorkflowView.js [hunk 1]
```diff
@@ -1219,9 +1219,9 @@ const WorkflowView = _.flow(
     async uploadJson(key, file) {
       try {
         const rawUpdates = JSON.parse(await readFileAsText(file));
-        const updates = _.mapValues((v) => (_.isString(v) && v.match(/\${(.*)}/) ? v.replace(/\${(.*)}/, (_, match) => match) : JSON.stringify(v)))(
-          rawUpdates
-        );
+        const updates = _.mapValues((v) =>
```
Comment (msilva-broad) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: my first questions is, where are the unit tests?   Looking at this file it's HUGE and complicated... as a smaller win, this functionality should get broken out to a workflow-view-utils.ts file, give it a bit of TS love, and unit test the code this PR is specifically changing.... 

This file needs alot of TS, unit testing, and "breaking up into smaller peices" work.  But let's start small :)

Comment (rickymagner) [reply src/pages/workspaces/workspace/workflows/WorkflowView.js]: Definitely agree generally with the assessment the file seems a bit monolithic as an outsider. But as a non-DSP Eng Terra user (DSP Methods), having experienced this bug a few times just this week alone while the PR has been open, I'd appreciate being able to merge this as a quickfix if you'd be willing to approve the PR. Then maybe the workflows team can add a ticket on refactoring/adding tests for the file?

Comment (msilva-broad) [reply src/pages/workspaces/workspace/workflows/WorkflowView.js]: I reached out to @mspector on Workflows team.  He and/or team will reach out on Monday and help get this to the finish line.

Comment (cjllanwarne) [reply src/pages/workspaces/workspace/workflows/WorkflowView.js]: Hi @rickymagner I just quickly extracted this method into a utils class and added a framework for a unit test and pushed to your branch. If you could add a couple of tests there, that would be great.

Thanks!

Comment (rickymagner) [reply src/pages/workspaces/workspace/workflows/WorkflowView.js]: Thanks for providing the template! It was really helpful given I haven't written JS tests before.

I just pushed a commit with 3 tests filled in: one "normal" parsing of a string, one parsing of a multi-line array, and one parsing of a badly formatted JSON that doesn't follow the Terra convention and gets stringified.

**Review by mspector — APPROVED** [review]

**Pull request merged**